src/passes/InstrumentMemory.cpp

```diff
@@ -300,14 +341,61 @@ struct InstrumentMemory : public WalkerPass<PostWalker<InstrumentMemory>> {
   }
 
 private:
-  Index id;
+  Index id = 0;
+  InstructionFilter _filter;
 
   void addImport(Module* curr, Name name, Type params, Type results) {
     auto import = Builder::makeFunction(name, Signature(params, results), {});
     import->module = ENV;
     import->base = name;
     curr->addFunction(std::move(import));
   }
+
+  template<typename T> bool checkExpression(T* expr) {
+    if (_filter.has_value()) {
+      return _filter->count(expressionToString(expr)) > 0;
+    }
+    return true;
+  }
+
+  std::string loadStoreTypeToString(Type type) { return type.toString(); }
+
+  std::string expressionToString(Load* expr) {
+    const auto suffix =
+      (expr->bytes == expr->type.getByteSize()
+         ? ""
+         : std::to_string(expr->bytes * 8) + "_" + (expr->signed_ ? 's' : 'u'));
+
+    return expr->type.toString() + ".load" + suffix;
+  }
+
+  std::string expressionToString(Store* expr) {
+    const auto suffix = expr->bytes == expr->valueType.getByteSize()
+                          ? ""
+                          : std::to_string(expr->bytes * 8);
+
+    return expr->valueType.toString() + ".store" + suffix;
+  }
+
+  std::string expressionToString(StructGet* expr) { return "struct.get"; }
+  std::string expressionToString(StructSet* expr) { return "struct.set"; }
+  std::string expressionToString(ArrayGet* expr) { return "array.get"; }
+  std::string expressionToString(ArraySet* expr) { return "array.set"; }
+  std::string expressionToString(MemoryGrow* expr) { return "memory.grow"; }
```

Comment:
Can we use `getExpressionName()` here? That uses internal IR names, but still might be enough. That is, do we really want to differentiate `i32.load` from `i64.load8_s` etc?

Alternatively, if we do want the precise names in wat text, how about using `ostream << ShallowExpression`, which prints out the wat text? We would just need to remove `(` etc.

But even simpler would be
```cpp
CHECK_EXPRESSION(curr, "memory.copy");
```
etc., that is, just write the text name there? (assuming we don't want to differentiate between operations inside a class, so Load refers to all `*.load*`).

Reply:
I personally don't need such granularity (in fact, at this stage I only need `memory.grow`). I like the idea with passing a string to `CHECK_EXPRESSION` (we don't even need the `curr` being passed there). I've updated the code, please let me know if you have any other suggestions.